ppp_datamodel/utils.py

```diff
@@ -9,3 +9,33 @@ def predicate(node, childs):
         else:
             return any(childs.values())
     return tree.fold(predicate)
+
+def isincluded(tree1,tree2):
+    """
+        Return True if and only if tree1 is included in tree2.
+    """
+    if isinstance(tree1,Resource):
+        tree1=List([tree1])
+    if isinstance(tree2,Resource):
+        tree2=List([tree2])
+    if type(tree1) != type(tree2):
+        return False
+    if isinstance(tree1,Missing):
+        return True
+    if isinstance(tree1,List):
+        return set(tree1.list).issubset(set(tree2.list))
+    if isinstance(tree1,Triple):
+        return  isincluded(tree1.subject,tree2.subject) and\
+            isincluded(tree1.predicate,tree2.predicate) and\
+            isincluded(tree1.object,tree2.object)
+    if isinstance(tree1,Sort):
+        return tree1.predicate == tree2.predicate and isincluded(tree1.list,tree2.list)
+    if isinstance(tree1,First) or isinstance(tree1,Last) or isinstance(tree1,Exists):
+        return isincluded(tree1.list,tree2.list)
+    # Intersection, Union, And, Or
```

Comment:
You can't be sure this part won't break when we will add more classes.
Please put it in a block that tests if it is any of these classes, and add an else clause.